auth-mailchimp-sync/functions/src/index.ts

```diff
@@ -95,5 +99,83 @@ export const removeUserFromList = functions.handler.auth.user.onDelete(
     } catch (err) {
       logs.errorRemoveUser(err);
     }
-  }
-);
+  });
+
+export const addExistingUsersToList = functions.tasks
+  .taskQueue()
+  .onDispatch(async (data) => {
+    const runtime = getExtensions().runtime();
+    if (!config.doBackfill) {
+      await runtime.setProcessingState(
+        "PROCESSING_COMPLETE",
+        "No existing documents imported into BigQuery."
+      );
+      return;
+    }
+    const nextPageToken = data.nextPageToken;
+    const pastSuccessCount = (data["successCount"] as number) ?? 0;
+    const pastErrorCount = (data["errorCount"] as number) ?? 0;
+
+    const res = await getAuth().listUsers(100, nextPageToken);
+    const mailchimpPromises = res.users.map(async (user) => {
+      const { email, uid } = user;
+      if (!email) {
+        logs.userNoEmail();
+        return;
+      }
+
+      try {
+        logs.userAdding(uid, config.mailchimpAudienceId);
+        const results = await mailchimp.post(
+          `/lists/${config.mailchimpAudienceId}/members`,
+          {
+            email_address: email,
+            status: config.mailchimpContactStatus,
+          }
+        );
+        logs.userAdded(
+          uid,
+          config.mailchimpAudienceId,
+          results.id,
+          config.mailchimpContactStatus
+        );
+      } catch (err) {
+        logs.errorAddUser(err);
+      }
+    });
+    const results = await Promise.allSettled(mailchimpPromises);
+    const newSucessCount =
+      pastSuccessCount + results.filter((p) => p.status === "fulfilled").length;
+    const newErrorCount =
+      pastErrorCount + results.filter((p) => p.status === "rejected").length;
+    if (res.pageToken) {
+      const queue = getFunctions().taskQueue(
+        "fsimportexistingdocs",
```

Comment:
Has this been tested? Because the name of this function is not fsimportexistingdocs... so I don't think you'll find the taskQueue...

Reply:
This extension is not tested yet, since i did not realize mailchimp had a free plan.

Reply:
Tested it out, and it is working after a few changes:
<img width="1439" alt="Screen Shot 2022-11-18 at 3 27 33 PM" src="https://user-images.githubusercontent.com/4635763/202819973-40c8758d-4acf-42aa-a496-a34332840021.png">